### PR TITLE
Add automated releasing

### PR DESCRIPTION
**Pull request by oceanc80**
Add automated build/release on updates to the main branch and any new tags

Closes https://github.com/operator-framework/combo/issues/96

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          echo GORELEASER_ARGS="--rm-dist --skip-validate" >> $GITHUB_ENV
+        elif [[ $GITHUB_REF == refs/pull/* ]]; then
+          # PR build.
+          echo IMAGE_TAG="pr-$(echo "${GITHUB_REF}" | sed -E 's|refs/pull/([^/]+)/?.*|\1|')" >> $GITHUB_ENV
+        else
+          echo IMAGE_TAG="$(git describe --tags --always)" >> $GITHUB_ENV
+        fi
+
+    - name: Run goreleaser
+      run: make release
```
Comment (timflannagan) [inline .github/workflows/release.yaml:55]: I think we'd need to add this release target to the root Makefile, or use the first-class goreleaser github action.

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - "{{ .Env.IMAGE_REPO }}:{{ .Env.IMAGE_TAG }}-s390x"
+checksum:
+  name_template: 'checksums.txt'
+snapshot:
+  name_template: "{{ incpatch .Version }}-next"
+changelog:
+  use: github-native
+  skip: $DISABLE_RELEASE_PIPELINE
+release:
+  disable: $DISABLE_RELEASE_PIPELINE
```
Comment (timflannagan) [inline .goreleaser.yaml:73]: Running goreleaser directly using this .goreleaser.yaml configuration would be problematic as there's no layer that sits between evaluating the value of this variables, and the source of truth configuration file that goreleaser relies on when running it's tooling. We can either pivot to the rukpak implementation that's centered around checking a goreleaser template file, and using envsubst to evaluate these values at runtime and produces a static .goreleaser.yaml configuration file, or disable the release/changelog pipelines explicitly.

The latter implementation would require further updates to the workflow such that automation creates a release for us during tag creation triggers.

File: .goreleaser.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+env:
+- GOPROXY=https://proxy.golang.org|direct
+- GO111MODULE=on
+- CGO_ENABLED=0
+- PKG=github.com/operator-framework/combo/pkg/version
+- KUBERNETES_VERSION=v0.22.2
```
Comment (timflannagan) [inline .goreleaser.yaml:6]: This feels potentially brittle. Can we try and use the Makefile as the source of truth here, and export any variable that sets the Kubernetes version (if that variable even exists today)?

**Comment by timflannagan** [comment]
I had one more comment but I think we're getting close.

File: .goreleaser.yaml (hunk 3 above)
Comment (timflannagan) [reply .goreleaser.yaml:6]: Yup that's more-or-less what I had in mind.

Comment (oceanc80) [reply .goreleaser.yaml:6]: I think so. I think one of the other variables is exported that way already.

File: Makefile [hunk 4]
```diff
@@ -110,3 +110,15 @@ run-local: build-local-container load-image deploy ## Run Combo on local environ
...
+##################
+# Release Targets#
+##################
+export DISABLE_RELEASE_PIPELINE ?= true
+substitute:
+	envsubst < .goreleaser.template.yml > .goreleaser.yml
+
+release: GORELEASER ?= goreleaser
+release: GORELEASER_ARGS ?= --snapshot --rm-dist
+release: goreleaser substitute
```
Comment (timflannagan) [inline Makefile:123]: I think we'd need to remove the `goreleaser` target as a prerequisite target as that target doesn't appear to exist?

Comment (oceanc80) [reply Makefile]: Oops... missed that in the copy/paste clean up.

**Review by timflannagan — APPROVED** [review]

**Review by tylerslaton — APPROVED** [review]
Nice! This looks great. I think the only thing we need now is credentials. @joelanford, would you mind setting those up before we merge?

**Comment by tylerslaton** [comment]
Just thought of a follow-up for this. We'll probably want to create a Bundle format that we can submit for both OLM and RukPak. Not required for this PR but I'll create a follow-up.

**Comment by oceanc80** [comment]
> Just thought of a follow-up for this. We'll probably want to create a Bundle format that we can submit for both OLM and RukPak. Not required for this PR but I'll create a follow-up.

@tylerslaton Does [this issue](https://github.com/operator-framework/combo/issues/84) cover that?

**Comment by tylerslaton** [comment]
@oceanc80 It sure does, good catch!

**Pull request merged**